src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1601,7 +1619,24 @@ def _set_num_frames(return_token_timestamps, generation_config, kwargs):
                     "Model generation config has no `alignment_heads`, token-level timestamps not available. "
                     "See https://gist.github.com/hollance/42e32852f24243b748ae6bc1f985b13a on how to add this property to the generation config."
                 )
-            generation_config.num_frames = kwargs.pop("num_frames", None)
+            if "num_frames" in kwargs:
+                generation_config.num_frames = kwargs.pop("num_frames")
+                if isinstance(generation_config.num_frames, torch.Tensor):
+                    generation_config.num_frames = generation_config.num_frames.cpu()
+                else:
+                    generation_config.num_frames = torch.tensor(generation_config.num_frames)
+
+                warnings.warn(
```

Comment:
why use warning and logger hahha let'salso only warn once and warn as little as possible